assets/gaussholder.js

```diff
@@ -82,25 +105,14 @@ window.Gaussholder = (function (header) {
 		element.style.height = final[1] + 'px';
 
 		// ...then recalculate based on what it actually renders as
-		var original = [ final[0], final[1] ];
-		if ( element.width < final[0] ) {
-			// Rescale, keeping the aspect ratio
-			final[0] = element.width;
-			final[1] = final[1] * ( final[0] / original[0] );
-		} else if ( element.height < final[1] ) {
-			// Rescale, keeping the aspect ratio
-			final[1] = element.height;
-			final[0] = final[0] * ( final[1] / original[1] );
-		}
-
-		// Set dimensions, _again_
-		element.style.width = final[0] + 'px';
-		element.style.height = final[1] + 'px';
+		calculateDimensionStyle( element );
 
+		// Schedule an observer to update on any rendering changes.
 		render(canvas, element.dataset.gaussholder.split(','), final, function () {
 			// Load in as our background image
 			element.style.backgroundImage = 'url("' + canvas.toDataURL() + '")';
 			element.style.backgroundRepeat = 'no-repeat';
+			setTimeout( function() { calculateDimensionStyle( element ); }, 200 );
```

Comment:
Could we use the image element load event perhaps? This still seems like it could be brittle

Reply:
This is already happening in the image's `onload` event handler. The timeout here is just trying to work around some issues where the dimensions weren't calculated properly, because apparently the "onload" event fires when the file load is complete, but before the DOM rendering is necessarily complete. 

I could probably remove the 200ms timeout and just leave it in a timeout or a requestAnimationFrame to make sure the image renders on screen before measuring it. You're right, this does feel brittle - I'll clean it up and see what I can do to make the callback happen as soon as possible.

Reply:
Ah ok, so if this is really just adding a small extra buffer right now before the blurred image is getting swapped out then I think it's ok. I'll test it out on local and see what happens to see if I have any other ideas.